File: Task3/Task3.php
<?php
function sumAndProductOfDigits($n) {
    $sum = 0;
    $product = 1;
    while ($n > 0) {
        $digit = $n % 10;
        $sum += $digit;
        $product *= $digit;
        $n = intval($n / 10);
    }
    return [$sum, $product];
}

echo "Кол-во чисел: ";
$n = intval(fgets(STDIN));
$indices = ""; // Строка для хранения индексов

for ($i = 0; $i < $n; $i++) {
    echo "$i число: ";
    $num = intval(fgets(STDIN));
    list($sum, $product) = sumAndProductOfDigits($num);
    if ($sum < $product) {
        $indices .= $i . " "; // Добавляем индекс в строку
    }
}
echo trim($indices) . "\n"; // Выводим индексы в конце
?>

File: Task3/Task3.rb
def SumAndProductOfDigits(n)
  sum = 0
  product = 1
  while n > 0
    digit = n % 10
    sum += digit
    product *= digit
    n /= 10
  end
  return sum, product
end

print "Кол-во чисел: "
n = gets.to_i
indices = ""

(0...n).each do |i|
  print "#{i} число: "
  num = gets.to_i
  sum, product = SumAndProductOfDigits(num)
  if sum < product
    indices += "#{i} "
  end
end

puts indices